feature_details.md

```diff
@@ -0,0 +1,174 @@
+
+It’s conceivable that support for animatable properties will be added incrementally and not all will be supported across UAs. This is still an open question and one option could be for the API to allow feature detection to detect which properties are supported by the UA. Using an unsupported property on a shared element has undefined behaviour.
+
+#### Animation Configuration
+The following fields are used in a dictionary, referred to as `ElementTransition`, to configure a single enter/exit or paired transition :
+
+* `transitionType` : This is an enum to provide a predefined set of animation patterns for enter/exit transitions. These can be explicit patterns like `slide-left` (does a translation in the specified direction) which provide convenient defaults for common use-cases. For instance, sliding a header/footer offscreen. Or higher level abstractions like `previous`/`next` which allow the UA to decide the appropriate animation (for example based on the idiomatic pattern for the native OS).
+
+* `duration` : The total length for this transition.
+
+* `delay` : A Document transition establishes a universal timeline for animations spanning multiple elements. This field indicates the delay in starting animations for this transition on this universal timeline.
+
+* `element` : A reference to the [Element](https://dom.spec.whatwg.org/#interface-element) in the new Document.
+
+* `sharedElement` : A reference to an element in the previous Document, explained in Referencing Elements (#referencing-elements).
+
+If one of `sharedElement` or `element` is set, this is an enter or exit transition respectively. The animations for each animatable property are defined based on `transitionType`. If both `sharedElement` and `element` are set, this is a paired transition. This transition first creates a combined image using painted content from both elements based on the `content` animation specification under [Animatable Properties](#animatable-properties). Other animatable properties are then applied to this combined output.
+
+An [open request](https://github.com/WICG/shared-element-transitions/issues/28) here is the ability to add low-level customization to these animations by specifying a delay, duration, easing curve, keyframes similar to the [Web Animation API](https://developer.mozilla.org/en-US/docs/Web/API/Web_Animations_API). A potential option to provide this capability in future iterations is to add a `keyframes` field to this dictionary which supports property types listed under [Animatable Properties](#animatable-properties). The syntax is defined under [Processing a keyframes argument](https://drafts.csswg.org/web-animations-1/#processing-a-keyframes-argument) in Web Animations API. It’s worth mentioning that the keyframes syntax already has a concept of an implicit start/end keyframe which could map to the start/end state for each property computed in the previous/new Document respectively.
```

Comment:
> A potential option to provide this capability in future iterations is to add a `keyframes` field to this dictionary which supports property types

This seems like we would have to reimplement web animations to a degree, just in a different spot. Is this really the request?

Reply:
This is indeed the request, yes.

The idea of lower level customization to control the following aspects sounds totally reasonable to me :
- Easing curve for each property, a separate value for opacity vs transform for example.
- Delay to stagger the animations. For example the opacity animation starts after the transform animation has finished.
- Separate duration for each property.

But it's still worth discussing whether specifying values for each property is built on top of CSS transitions (animating between 2 states) or keyframes as specified in Web Animations. I don't think we need to resolve this question at this stage, but we should ensure the API can be extended to accomodate this request.